File: src/game/tools/rpc/output/wrap/HeroCommentListDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type HeroCommentListDataWrap struct {
	
	CommentsList []*HeroCommentDataWrap // 评论列表
	rpc.Wrapper
}

func (w *HeroCommentListDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.CommentsList = make([]*HeroCommentDataWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.CommentsList); i++ {
		w.CommentsList[i] = new(HeroCommentDataWrap)
		w.CommentsList[i].Decode(pck)
		
	}
	
	return w
}

func (w *HeroCommentListDataWrap)Encode(pck *rpc.Packet) {
	
	if w.CommentsList == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.CommentsList)))
    	for i := 0; i < len(w.CommentsList); i++ {
    		w.CommentsList[i].Encode(pck);
			
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/HeroCommentListDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class HeroCommentListDataWrap : IWrapper
	{	
		public HeroCommentDataWrap[] CommentsList; // 评论列表
		
		public void Decode(Packet pck)
		{	
			CommentsList = new HeroCommentDataWrap[pck.GetShort()];
			for (int i = 0; i < CommentsList.Length; i++)
			{
				CommentsList[i] = new HeroCommentDataWrap();
				CommentsList[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (CommentsList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)CommentsList.Length);
	        	for(int i = 0; i < CommentsList.Length; i++)
	        	{
	        		CommentsList[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}